src/types_ext.cc

```diff
@@ -263,6 +263,10 @@ namespace UserDbReg{
     return {};
   }
 
+  string file_name(const T& t) {
+    return t.file_path().u8string();
```

Comment:
Is it expected that `file_name(db)` should return UTF-8 encoded path (instead of `CP_ACP` on Windows)?

UTF-8 string is appropriate if the returned string path is used to extract schema ID, or passed to librime code as path.

The returned string path might need to be converted to ANSI encoding before given to lua file-system functions.

Reply:
@hchunhui 
@shewer 
PTAL

Reply:
AFAIK, lua uses ANSI encoding path.

Reply:
OK. Let's make it ANSI encoded.

Note this is the original build error because the C++ class no longer has the getter `file_name`.

A reminder from the summary of the refactor in librime:

`file_name`'s output, ANSI encoded path, is suitable for accessing the file system using Lua binding to system functions.
If any wants to get schema ID, dictionary ID from the file name, it has to be first converted to UTF-8 string, in order to support path with Chinese characters on Windows.
When constructing a path from schema ID or UTF-8 encoded string read from config, conversion is also needed.

Reply:
Yes, ANSI encoded path is subtle, but this is lua's limitation. We need a custom source code of lua if we want UTF-8 encoded path.